phoenix-tools/src/main/java/org/apache/phoenix/schema/SchemaSynthesisProcessor.java

```diff
@@ -0,0 +1,261 @@
+    public static final String
+            ENTITY_NAME_IN_BASE_AND_ALTER_DDL_DON_T_MATCH =
+            "Entity name in base and alter DDL don't match";
+    private final String baseDDLFile;
+    private final String alterDDLFile;
+
+    public SchemaSynthesisProcessor(String baseDDLFile, String alterDDLFile) {
+        this.baseDDLFile = baseDDLFile;
+        this.alterDDLFile = alterDDLFile;
+    }
+
+    @Override
+    public String process() throws Exception {
+        String baseDDL = getQueriesFromFile(baseDDLFile).get(0);
+        List <String> alterDDL = getQueriesFromFile(alterDDLFile);
+        for(String s : alterDDL) {
+            baseDDL = synthesize(baseDDL, s);
+        }
+        return baseDDL;
+    }
+
+    private String synthesize(String baseDDL, String alterDDL) throws Exception {
+        BindableStatement createStatement = new SQLParser(baseDDL).parseStatement();
+        BindableStatement alterStatement = new SQLParser(alterDDL).parseStatement();
+        if (createStatement instanceof CreateTableStatement) {
+            CreateTableStatement newCreateStmt = null;
+            CreateTableStatement createStmt = (CreateTableStatement) createStatement;
+            if (alterStatement instanceof AddColumnStatement) {
+                newCreateStmt =
+                        getCreateTableStatement((AddColumnStatement) alterStatement, createStmt);
+            } else if(alterStatement instanceof DropColumnStatement) {
+                newCreateStmt =
+                        getCreateTableStatement((DropColumnStatement) alterStatement, createStmt);
+            }
+            return getCreateTableSQL(newCreateStmt);
+        } else if (createStatement instanceof CreateIndexStatement) {
+            CreateIndexStatement newCreateIndexStmt =
+                    getCreateIndexStatement((CreateIndexStatement) createStatement, alterStatement);
+            return getCreateIndexSQL(newCreateIndexStmt);
+        }
+        return "";
+    }
+
+    private CreateIndexStatement getCreateIndexStatement(CreateIndexStatement createStatement,
+            BindableStatement alterStatement) throws Exception {
+        CreateIndexStatement newCreateIndexStmt = null;
+        String tableName = createStatement.getIndexTableName().toString();
+        String tableNameInAlter = ((AddColumnStatement)alterStatement).getTable().toString().trim();
+        sanityCheck(tableName, tableNameInAlter);
+        AddColumnStatement addStmt = (AddColumnStatement) alterStatement;
+        if (addStmt.getColumnDefs() == null) {
+            ListMultimap<String, Pair<String, Object>>
+                    finalProps =
+                    getEffectiveProperties(addStmt, createStatement.getProps());
+            newCreateIndexStmt = new CreateIndexStatement(createStatement, finalProps);
+        }
+        return newCreateIndexStmt;
+    }
+
+    private CreateTableStatement getCreateTableStatement(DropColumnStatement alterStatement,
```

Comment:
nit: up to now we've been doing parameters as Create, Alter, and this switches to Alter, Create. Good to be consistent.